airbyte-integrations/connectors/source-convex/source_convex/source.py

```diff
@@ -0,0 +1,141 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import IncrementalMixin, Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+def convex_url_base(instance_name) -> str:
+    return f"https://{instance_name}.convex.cloud"
+
+
+# Source
+class SourceConvex(AbstractSource):
+    def _json_schemas(self, config) -> requests.Response:
+        instance_name = config["instance_name"]
+        access_key = config["access_key"]
+        url = f"{convex_url_base(instance_name)}/api/0.2.0/json_schemas"
```

Comment:
what do we want to use as the `version`? i assume we don't want to leave it as 0.2.0

Reply:
Hmm... since we have dedicate client, I think it might make sense to have a constant at the top of this file. Like "PROTOCOL_VERSION = 1.0". Then we change all egress_routes to be "/api/:protocol_version". This way it is clear that this is different than the convex react client.

This is if you plan to use the same egress endpoints across airbyte and fivetran. If not this can be named "/api/:airbyte_exporter_version".

Reply:
resolved with tom b that we won't version this api at all in the url